tests/non_matching/step-70.cc

```diff
@@ -0,0 +1,1983 @@
+            mpi_communicator);
+
+          relevant_tracer_particle_displacements = tracer_particle_velocities;
+
+          tracer_particle_handler.set_particle_positions(
+            relevant_tracer_particle_displacements);
+        }
+
+        // Using these new locations, we can then assemble the Stokes system and
+        // solve it.
+        assemble_stokes_system();
+        assemble_nitsche_restriction();
+        solve();
+
+        // With the appropriate frequencies, we then write the information of
+        // the solid particles, the tracer particles, and the fluid domain into
+        // files for visualization, and end the time step by adapting the mesh.
+        if (cycle % par.output_frequency == 0)
+          {
+            output_results(output_cycle, time);
+            {
+              output_particles(tracer_particle_handler,
+                               "tracer",
+                               output_cycle,
+                               time);
+            }
+            {
+              output_particles(solid_particle_handler,
+                               "solid",
+                               output_cycle,
+                               time);
+            }
+            ++output_cycle;
+          }
+        if (cycle % par.refinement_frequency == 0 &&
+            cycle != par.number_of_time_steps - 1)
+          refine_and_transfer();
+      }
+  }
+
+} // namespace Step70
+
+
+// @sect3{The main() function}
+
+// The remainder of the code, the `main()` function, is standard, with the
+// exception of the handling of input parameter files. We allow the user to
+// specify an optional parameter file as an argument to the program. If
+// nothing is specified, we use the default file "parameters.prm", which is
+// created if non existent. The file name is scanned for the string "23"
+// first, and "3" afterwards. If the filename contains the string "23", the
+// problem classes are instantiated with template arguments 2 and 3
+// respectively. If only the string "3" is found, then both template arguments
+// are set to 3, otherwise both are set to 2.
+//
+// If the program is called without any command line arguments (i.e.,
+// `argc==1`), then we just use "parameters.prm" by default.
+int
+main(int argc, char *argv[])
+{
+  using namespace Step70;
+  using namespace dealii;
+
+  auto          init = Utilities::MPI::MPI_InitFinalize(argc, argv, 1);
+  MPILogInitAll log(true);
+
+  deallog.depth_console(1);
+  try
+    {
+      std::string prm_file;
+      if (argc > 1)
+        prm_file = argv[1];
+      else
+        prm_file = SOURCE_DIR "/step-70.in.prm";
+
+      if (prm_file.find("23") != std::string::npos)
+        {
+          StokesImmersedProblemParameters<2, 3> par;
+          ParameterAcceptor::initialize(prm_file);
+
+          StokesImmersedProblem<2, 3> problem(par);
+          problem.run();
+        }
+      else if (prm_file.find('3') != std::string::npos)
```

Comment:
@bangerth I don't know where this code comes from, but this at least explains the problem: The PR has the number '15370', and the CI MPI tester creates a directory for running this test including the folder name. This then appears in `prm_file` that then triggers the code to run in 3d, as far as I can see. Can you try to remove this brittle part to read off the dimension from the `main()` function, then we can merge this.

I check the test on my machine, where it succeeds.

Reply:
Wow, that's wild!

I've fixed this. Let's see what happens. The code is from the original step-70, as it exists in our examples directory. I'll open an issue for us to fix.

Reply:
See #15453.